vm_supervisor/migrations/env.py

```diff
@@ -0,0 +1,70 @@
+from logging.config import fileConfig
+
+from alembic import context
+from vm_supervisor.conf import make_db_url
+from sqlalchemy import create_engine
+
+# # this is the Alembic Config object, which provides
+# # access to the values within the .ini file in use.
+# config = context.config
+#
+# # Interpret the config file for Python logging.
+# # This line sets up loggers basically.
+# if config.config_file_name is not None:
+#     fileConfig(config.config_file_name)
+
+# Auto-generate migrations
+from vm_supervisor.metrics import Base
+target_metadata = Base.metadata
+
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+# my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+
+
+def run_migrations_offline() -> None:
+    """Run migrations in 'offline' mode.
+
+    This configures the context with just a URL
+    and not an Engine, though an Engine is acceptable
+    here as well.  By skipping the Engine creation
+    we don't even need a DBAPI to be available.
+
+    Calls to context.execute() here emit the given string to the
+    script output.
+
+    """
+    url = make_db_url()
+    context.configure(
```

Comment:
New to Alembic. Is this call purely stateful ?

Reply:
Yes I think so. Don't fret about the details too much here, most of this file is automatically generated.